perseo-main/src/main/java/com/telefonica/iot/perseo/RulesManager.java

```diff
@@ -114,33 +127,52 @@ public static synchronized Result make(EPServiceProvider epService, String text)
+                String dId = deployment.getDeploymentId();
+                statement = deployment.getStatements()[0];
+                logger.debug(String.format("statement json: %s", Utils.Statement2JSONObject(
+                                                                                            statement,
+                                                                                            epa,
+                                                                                            dId
+                                                                                            )));
                 statement.addListener(new GenericListener());
             } else {
-                String oldEpl = prevStmnt.getText();
+                String oldEpl = prevStmnt.getProperty(StatementProperty.EPL).toString();
                 logger.debug(String.format("old epl: %s", oldEpl));
                 if (!newEpl.equals(oldEpl)) {
                     logger.debug(String.format("found changed statement: %s",name));
-                    prevStmnt.destroy();
+                    try {
+                        epa.undeploy(prevStmnt.getDeploymentId());
+                    } catch (EPUndeployException ex) {
+                        throw new RuntimeException(ex);
+                    }
                     logger.debug(String.format("deleted statement: %s",name));
-                    statement = epService.getEPAdministrator().createEPL(newEpl, name);
+
+                    EPDeployment deployment = Utils.compileDeploy(epService, newEpl, name);
+                    String dId = deployment.getDeploymentId();
+                    statement = deployment.getStatements()[0];
                     logger.debug(String.format("re-created statement: %s", name));
-                    logger.debug(String.format("statement json: %s", Utils.Statement2JSONObject(statement)));
+                    logger.debug(String.format("statement json: %s", Utils.Statement2JSONObject(
+                                                                                                statement,
+                                                                                                epa,
+                                                                                                dId
+                                                                                                )));
                     statement.addListener(new GenericListener());
                 } else {
                     logger.debug(String.format("found repeated statement: %s", name));
                     statement = prevStmnt;
                 }
             }
             return new Result(HttpServletResponse.SC_OK,
-                    Utils.Statement2JSONObject(statement).toString());
+                              Utils.Statement2JSONObject(statement, epa, statement.getDeploymentId()).toString());
```

Comment:
Over indentation here?

Reply:
fixed in https://github.com/telefonicaid/perseo-core/pull/176/commits/24494fc64018425accd9080f0efee0126e5d7725